File: Module 2/C# II/homework_5_c_sharp_due_30.11.2016/15. Replace tags/ReplaceTags.cs
/*
Replace tags

Description
Write a program that replaces in a HTML document given as string all the tags <a href="URL">TEXT</a> 
with corresponding markdown notation [TEXT](URL).

Input
On the only input line you will receive a string

Output
Print the string with replaced tags

Constraints
Time limit: 0.1s
Memory limit: 16MB

Sample tests

Input
<p>Please visit <a href="http://academy.telerik.com">our site</a> to choose a training course. Also visit <a href="www.devbg.org">our forum</a> to discuss the courses.</p>

Output
<p>Please visit [our site](http://academy.telerik.com) to choose a training course. Also visit [our forum](www.devbg.org) to discuss the courses.</p>
*/

using System;
using System.Text.RegularExpressions;

class ReplaceTags
{
    static void Main()
    {
        string inputStr = Console.ReadLine();
        string result = Regex.Replace(inputStr, @"<a.href=""([^>]*)"">([^<]*(?:(?!</a)<[^<]*)*)</a>", "[$2]($1)");
        Console.WriteLine(result);
    }
}

File: Module 1/JS I - Fundamentals/homework_7_due_18.01.2017/problem_08/ReplaceTags.js
// Problem 08. Replace tags - BGCoder format
function solve(args) {
	let input =  args[0].split('<a href='),
		result = '';
	
	for (let i = 0; i < input.length; i += 1) {
		
		let part = input[i];
		if (part[0] === '"') {
			let url = ('(' + part.split('</a>')[0].split('>')[0].split('"')[1] + ')'),
				linkText = ('[' + part.split('</a>')[0].split('>')[1] + ']'),
				ending = part.split('</a>')[1];
			
			part = (linkText + url + ending);
		}
		
		result += (part);
	}		
	
	return result;
}

var test = [ '<p>Please visit <a href="http://academy.telerik.com">our site</a> to choose a training course. Also visit <a href="www.devbg.org">our forum</a> to discuss the courses.</p>' ];
console.log(solve(test));